dali/test/python/test_operator_slice.py

```diff
@@ -701,3 +704,86 @@ def test_no_slice():
     for device in ['cpu', 'gpu']:
         for dtype in [types.UINT8, types.UINT16, types.FLOAT]:
             yield check_no_slice, device, dtype, batch_size, num_threads
+
+def check_dynamic_axes(device, batch_size, num_threads, use_negative):
+    def get_dynamic_axes():
+        options = [
+            np.array([0, 1], dtype=np.int32),
+            np.array([1, 0], dtype=np.int32),
+            np.array([0], dtype=np.int32),
+            np.array([1], dtype=np.int32),
+        ]
+        if use_negative:
+            options += [
+                np.array([-2, -3], dtype=np.int32),
+                np.array([-2, 0], dtype=np.int32),
+                np.array([-3, 1], dtype=np.int32),
+                np.array([0, -2], dtype=np.int32),
+                np.array([1, -3], dtype=np.int32),
+                np.array([-2], dtype=np.int32),
+                np.array([-3], dtype=np.int32),
+            ]
+
+        axes = []
+        rel_start = []
+        rel_shape = []
+        for _ in range(batch_size):
+            axes_choice = random.choice(options)
+            axes += [axes_choice]
+            rel_start += [np.random.uniform(0.0, 0.2, axes_choice.shape)]
+            rel_shape += [np.random.uniform(0.4, 0.6, axes_choice.shape)]
+        return axes, rel_start, rel_shape
+
+    @pipeline_def(batch_size=batch_size, num_threads=num_threads, device_id=0)
+    def make_pipe():
+        encoded, _ = fn.readers.caffe(path=caffe_db_folder, random_shuffle=False)
+        image = fn.decoders.image(encoded, device="cpu", output_type=types.RGB)
+        if device == 'gpu':
+            image = image.gpu()
+        axes, rel_start, rel_shape = fn.external_source(source=get_dynamic_axes, num_outputs=3)
+        rel_start = fn.cast(rel_start, dtype=types.FLOAT)
+        rel_shape = fn.cast(rel_shape, dtype=types.FLOAT)
```

Comment:
Do we need to cast here? Is it possible to set `dtype=np.float32` in `get_dynamic_axes`?